### PR TITLE
[Model Monitoring] Update `model_uri` when needed & edit `model_uri` to use `ModelArtifact.db_key` instead of the `ModelArtifact.key`

### PR DESCRIPTION
**Pull request by davesh0812**
Resolves  [ML-7608](https://iguazio.atlassian.net/browse/ML-7608), 
[ML-7714](https://iguazio.atlassian.net/browse/ML-7714)

[ML-7608]: https://iguazio.atlassian.net/browse/ML-7608?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ML-7714]: https://iguazio.atlassian.net/browse/ML-7714?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mlrun/serving/v2_serving.py [hunk 1]
```diff
@@ -554,13 +555,13 @@ def _init_endpoint_record(
     except mlrun.errors.MLRunNotFoundError:
         model_ep = None
     except mlrun.errors.MLRunBadRequestError as err:
-        logger.debug(
+        logger.info(
             f"Cant reach to model endpoints store, due to  : {err}",
         )
```
Comment (jond01) [inline mlrun/serving/v2_serving.py]:
```suggestion
        logger.info(
            "Cannot get the model endpoints store", err=mlrun.errors.err_to_str(err)
        )
```

File: mlrun/serving/v2_serving.py [hunk 2]
```diff
@@ -586,28 +587,35 @@ def _init_endpoint_record(
...
+                mlrun.common.schemas.model_monitoring.ModelMonitoringMode.enabled
+                if model.context.server.track_models
+                else mlrun.common.schemas.model_monitoring.ModelMonitoringMode.disabled
+            )
+        if attributes:
+            db = mlrun.get_run_db()
+            db.patch_model_endpoint(
+                project=project,
+                endpoint_id=uid,
+                attributes={"model_uri": model_ep.spec.model_uri},
```
Comment (jond01) [inline mlrun/serving/v2_serving.py]: Shouldn't it be:
```suggestion
                attributes=attributes,
```
?

**Review by jond01 — COMMENTED** [review]
What about the `_init_endpoint_record` function in `mlrun/serving/routers.py`?

In addition, I think unit-tests can help in the multiple cases here.

File: mlrun/model_monitoring/helpers.py [hunk 3]
```diff
@@ -322,3 +323,37 @@ def get_invocations_metric(project: str) -> ModelEndpointMonitoringMetric:
...
         full_name=get_invocations_fqn(project),
     )
+
+
+def enrich_model_endpoint_with_model_uri(
+    model_endpoint: ModelEndpoint,
+    model_obj: mlrun.artifacts.ModelArtifact,
+):
+    """
+    Enrich the model endpoint object with the model uri from the model object. Instead of using the model uri that
```
Comment (assaf758) [inline mlrun/model_monitoring/helpers.py]: this documentation is more suitable to the jira (where we describe the fix we are planning to do).
Here we should just describe what the function do.

Comment (davesh0812) [reply mlrun/model_monitoring/helpers.py]: @assaf758 , I just copy paste the function to different file.. but no problem I will change it :)

**Review by jond01 — APPROVED** [review]

**Review by assaf758 — APPROVED** [review]

**Pull request merged**